docs/02-guides/server-side-events/create-an-event.mdx

```diff
@@ -1,15 +1,135 @@
 ---
 title: Create an event
 description:
-  This guide explains how to create an event in Front-Commerce and dispatch it to external services.
+  In this guide, we will cover the creation of an emitter and how to implement
+  them in your pages, we will create an Emitter that Emit an event when a
+  customer reach a page called FAQ.
 sidebar_position: 1
 ---
 
 <p>{frontMatter.description}</p>
 
-:::info WIP
+## What is an event emitter
 
-- Create an event emitter
-- Emit an event from your code
+The events are dispatched through the use of an event emitter. The role of the
+event emitter is to push events to the server event pipeline to be processed by
+the server.
 
-:::
+## Create a new route
+
+Create the following route in your project:
+
+```tsx title="app/routes/_main.faq.tsx"
+export default function FAQ() {
+  return (
+    <div>
+      <h1>FAQ</h1>
+    </div>
+  );
+}
+```
+
+And now test your route by starting your Front-Commerce server and navigate to
+[http://localhost:4000/faq](http://localhost:4000/faq).
+
+You should now see your brand new page.
+
+## Create an Emitter class
+
+Let's create a file in our project, called `FAQPageReached.ts`:
+
+```tsx title="app/server-events/events/FAQPageReached.ts"
+import { ServerEvent } from "@front-commerce/core/events";
+
+// Name of the event that will be dispatched to the external services
+const FAQPageReachedName = "FAQPageReachedEvent";
+
+// Object that will be sent to external services
+type FAQPageReachedEventType = {
+  question: string;
+};
+
+export default class FAQPageReachedEvent extends ServerEvent<FAQPageReachedEventType> {
```

Comment:
Is `ServerEvent` documented somewhere? reading this I can't find reference to `ServerEvent`, and seeing that you are passing in multiple params to the `super` it's unclear what those params represent.

I would say this needs to be documented under `API Reference` ->`@front-commerce/core` -> `Server Event`

Reply:
added